### DISCUSSION TITLE
An error occurs when bash is used to run graphrag_indexing.sh

### DISCUSSION THREAD

**Discussion by fuyuton** [mindverse/Second-Me#356] [Q&A]
## I have done a search for similar discussions

#348 

## environment

kvm machine 48GB RAM, QEMU Virtual CPU version 2.5+(CPU base: AMD Ryzen 7 PRO 4750GE with Radeon Graphics)

ubuntu 24.04

non Docker

no CUDA

secondme ver. 1.0.1

## parameter

Data Synthesis Mode: Low

Base Model: 0.5B

Learning Rate: 0.0001

Number of Epochs: 2

Concurrency Threads: 1

no GPU

## phenomenon

This error occurred.

```
backend.log:2025-05-14 05:40:13 [ERROR] trainprocess_service.py:316 - Map entity network failed: Command '['bash', '/opt/second-me/lpm_kernel/L2/data_pipeline/data_prep/scripts/graphrag_indexing.sh']' returned non-zero exit status 132.
```

However, when I set up secondme, I was told that zsh was required, so I set it as the default, but the error message said that bash was being called, which I thought was strange.

## What I tried

I manually ran graphrag_indexing.sh using bash, and of course I got an error.

```
secondme# bash /opt/second-me/lpm_kernel/L2/data_pipeline/data_prep/scripts/graphrag_indexing.sh
/opt/second-me/lpm_kernel/L2/data_pipeline/data_prep/scripts/graphrag_indexing.sh: line 1: 68387 Illegal instruction     (core dumped) graphrag index --config lpm_kernel/L2/data_pipeline/graphrag_indexing/settings.yaml --root lpm_kernel/L2/data_pipeline/graphrag_indexing --method standard --logger none
```

Then I manually ran graphrag_indexing.sh using zsh. There was silence. (Does this indicate success?)

```
secondme# zsh /opt/second-me/lpm_kernel/L2/data_pipeline/data_prep/scripts/graphrag_indexing.sh
secondme#
```

Since it seemed to be successful with zsh, I tried to find out where bash was being called from.

But the error message didn't give me any other information than that bash was being used, so I decided to look at the source code.

I followed the source code, but I couldn't figure out where this script was being called from.

But I knew it was bash. So every time I followed the source code, I would grep around at that level to search for it.

## result

I finally found the source code where bash is hardcoded.

https://github.com/mindverse/Second-Me/blob/master/lpm_kernel/L2/data.py#L577

I made this into zsh, ran make start, and started training from the frontend.

However, I got the same error as with bash.

```
2025-05-14 05:56:54 [ERROR] trainprocess_service.py:318 - Map entity network failed: Command '['zsh', '/opt/second-me/lpm_kernel/L2/data_pipeline/data_prep/scripts/graphrag_indexing.sh']' returned non-zero exit status 132.

```

## Question

It seems like it's hard for me to track it down any further at the moment.

What does exit status 132 stand for?

Has anyone had the same error and been able to resolve it?

One more thing. Why does the shebang in scripts/*.sh say bash? Is this correct?

**Comment by fuyuton** [comment]
I decided to investigate further.

The bottom line is that graphrag, and therefore secondme, cannot be used on virtual machines that use emulated CPUs.


It was a problem with graphrag.

When I ran graphrag as is, I got an illigal hardware instruction. This was the same whether I ran it in zsh or bash.

```
secondme# zsh
secondme# graphrag index --config lpm_kernel/L2/data_pipeline/graphrag_indexing/settings.yaml --root lpm_kernel/L2/data_pipeline/graphrag_indexing --method standard --logger none
zsh: illegal hardware instruction (core dumped)  graphrag index --config  --root lpm_kernel/L2/data_pipeline/graphrag_indexing
secondme# bash
root@secondme:/opt/second-me/lpm_kernel/L2/data_pipeline/data_prep/scripts# graphrag index --config lpm_kernel/L2/data_pipeline/graphrag_indexing/settings.yaml --root lpm_kernel/L2/data_pipeline/graphrag_indexing --method standard --logger none
Illegal instruction (core dumped)
```

Sorry for the trouble.
